android/src/main/java/com/ryanheise/audio_session/AndroidAudioManager.java

```diff
@@ -320,30 +325,32 @@ private boolean requestAudioFocus(List<?> args) {
             if (audioFocusRequest != null) {
                 return true;
             }
-            Map<?, ?> request = (Map<?, ?>)args.get(0);
-            AudioFocusRequestCompat.Builder builder = new AudioFocusRequestCompat.Builder((Integer)request.get("gainType"));
+            Map<?, ?> request = (Map<?, ?>) args.get(0);
+            AudioFocusRequestCompat.Builder builder = new AudioFocusRequestCompat.Builder((Integer) request.get("gainType"));
             builder.setOnAudioFocusChangeListener(focusChange -> {
                 if (focusChange == AudioManager.AUDIOFOCUS_LOSS) abandonAudioFocus();
                 invokeMethod("onAudioFocusChanged", focusChange);
             });
             if (request.get("audioAttributes") != null) {
-                builder.setAudioAttributes(decodeAudioAttributes((Map<?, ?>)request.get("audioAttributes")));
+                builder.setAudioAttributes(decodeAudioAttributes((Map<?, ?>) request.get("audioAttributes")));
             }
             if (request.get("willPauseWhenDucked") != null) {
-                builder.setWillPauseWhenDucked((Boolean)request.get("willPauseWhenDucked"));
+                builder.setWillPauseWhenDucked((Boolean) request.get("willPauseWhenDucked"));
             }
             audioFocusRequest = builder.build();
             int status = AudioManagerCompat.requestAudioFocus(audioManager, audioFocusRequest);
             boolean success = status == AudioManager.AUDIOFOCUS_REQUEST_GRANTED;
             if (success) {
                 registerNoisyReceiver();
+                registerScoReceiver();
```

Comment:
@ryanheise here is the java side, together with the `unregister` method.

sorry about all this "ident" changes, making it harder to read. let me know if you have a different configuration on JetBrains, so I can tweak it here.